### PR TITLE
Turf decals + path update helper

### PR DESCRIPTION
**Pull request by AnturK**
Simple decal system to replace turf stacking and icon state duplication.
Plus simple script to help with path updates that are bit more complicated that search&replace.

I'll probably replace regex in the script with simple parsing to make it more robust later.

### CODE REVIEW

**Comment by Cyberboss** [comment]
Brave man

**Comment by optimumtact** [comment]
Nice

File: code/game/objects/effects/decals/decal.dm [hunk 1]
```diff
@@ -9,3 +9,30 @@
...
+	icon = 'icons/turf/decals.dmi'
+	icon_state = "warningline"
+
+//in case we need some special decals
+/obj/effect/turf_decal/proc/get_decal()
+	return image(icon='icons/turf/decals.dmi',icon_state=icon_state,dir=dir,layer=TURF_LAYER)
+
+/obj/effect/turf_decal/initialize()
+	var/turf/T = loc
+	if(!istype(T)) //you know this will happen somehow
```
Comment (optimumtact) [inline code/game/objects/effects/decals/decal.dm]: crash("decal on non turf or turf in nullspace")

ends the proc and leaves a runtime with stacktrace

**Comment by optimumtact** [comment]
I wouldn't call this a feature since it only refactors existing code, if anyone wants to argue, you can strip out the script and make that somewhere else

**Comment by WJohn** [comment]
Thank you <3 <3

**Comment by AnturK** [comment]
I'm really not sure why some maps have the space behind the comment and some do not, i guess the mapmerge code must have changed at some point

**Comment by AnturK** [comment]
For anyone who wants to apply this the update script is here : http://pastebin.com/HAEMDf9F

**Comment by AnturK** [comment]
This should be ready for review, In-game methods of  placing these will come after freeze.

**Comment by Cyberboss** [comment]
@AnturK Add `Fixes #18888`

**Comment by AnturK** [comment]
It doesn't fix that (yet) this is just first step.

File: tools/mapmerge/path_update.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
+import re
+import os
+import argparse
+import map_helpers #why we don't have some generic package for these reee
```
Comment (xxalpha) [inline tools/mapmerge/path_update.py]: Because I never figured out how that shit works with python. 🐙

**Pull request merged**